JASP-Engine/JASP/R/selectionmodels.R

```diff
@@ -16,86 +16,109 @@
+  }
 }
-.SM_data_get           <- function(dataset, options){
-  if(!is.null(dataset)){
+.smGetData                 <- function(dataset, options) {
+  if (!is.null(dataset)) {
     return(dataset)
-  }else{
+  } else {
     return(.readDataSetToEnd(columns.as.numeric = c(
       options[["input_ES"]],
-      options[["input_SE"]],
-      if(options[["input_p"]] != "") options[["input_p"]]
+      if (options[["input_SE"]] != "") options[["input_SE"]],
+      if (options[["input_N"]] != "")  options[["input_N"]],
+      if (options[["input_p"]] != "")  options[["input_p"]]
     )))
   }
 }
-.SM_data_check         <- function(jaspResults, dataset, options){
+.smCheckData               <- function(jaspResults, dataset, options) {
   
-
   dataset_old <- dataset
   dataset     <- na.omit(dataset)
   
   # store the number of missing values
-  if(nrow(dataset_old) > nrow(dataset)){
-    if(!is.null(jaspResults[["ommited_observations"]])){
+  if (nrow(dataset_old) > nrow(dataset)) {
+    if (!is.null(jaspResults[["ommited_observations"]])) {
       ommited_observations <- jaspResults[["ommited_observations"]]
-    }else{
+    } else {
       ommited_observations <- createJaspState()
-      ommited_observations$dependOn(c(.SM_dependencies))
+      ommited_observations$dependOn(c(.smDependencies))
       jaspResults[["ommited_observations"]] <- ommited_observations
     }
     ommited_observations$object <- nrow(dataset_old) - nrow(dataset)
   }
   
   .hasErrors(dataset               = dataset,
-             type                  = c("infinity", "observations", "variance"),
+             type                  = c("infinity", "observations"),
              observations.amount   = "< 2",
              exitAnalysisIfErrors  = TRUE)
   
-  .hasErrors(dataset               = dataset,
-             type                  = c("negativeValues"),
-             negativeValues.target = options[["input_SE"]],
-             exitAnalysisIfErrors  = TRUE)
+  if (options[["input_SE"]] != "")
+    .hasErrors(dataset               = dataset,
+               type                  = c("negativeValues"),
+               negativeValues.target = options[["input_SE"]],
+               exitAnalysisIfErrors  = TRUE)
+  
+  
+  if (options[["input_N"]] != "")
+    .hasErrors(dataset               = dataset,
+               type                  = c("negativeValues"),
+               negativeValues.target = options[["input_N"]],
+               exitAnalysisIfErrors  = TRUE)
+  
+  
+  if (options[["input_p"]] != "") 
+    if (any(dataset[, .v(options[["input_p"]])] < 0) || any(dataset[, .v(options[["input_p"]])] > 1))
+      JASP:::.quitAnalysis(gettextf("Error in %s: All p-values need to be between 0 and 1.", options[["input_p"]]))
+  
+  
+  
+  if (options[["input_ES"]] != "" && options[["measures"]] == "correlation")
+    if (any(dataset[, .v(options[["input_ES"]])] <= -1) || any(dataset[, .v(options[["input_ES"]])] >= 1))
```

Comment:
no, because correlations are transformed to either Cohen's d or Fisher's z for the fitting purposes and both of them would be undefined. Also, you can't practically get a correlation of |1|.